File: front-end/src/service/ExamHistoryService.ts
import AnswerUserRequest from "../model/request/AnswerUserRequest";
import ApiResponse from "../model/response/ApiResponse";
import ExamHistoryDetailsResponse from "../model/response/details/ExamHistoryDetailResponse";
import ExamHistoryResponse from "../model/response/ExamHistoryResponse";
import PageResponse from "../model/response/PageResponse";
import instance from "../util/apiInstance";
const basePath = "exam-result";
export default class ExamHistoryService {
  static readByExamId(id: number): Promise<ApiResponse<ExamHistoryResponse>> {
    return instance.get(`${basePath}/exam/${id}`);
  }
  static readById(
    id: number,
    data: AnswerUserRequest[]
  ): Promise<ApiResponse<ExamHistoryResponse>> {
    return instance.post(`${basePath}/${id}`, data);
  }
  static readDetailsById(
    id: number
  ): Promise<ApiResponse<ExamHistoryDetailsResponse>> {
    return instance.get(`${basePath}/${id}`);
  }
  static readPage(): Promise<ApiResponse<PageResponse<ExamHistoryResponse>>> {
    return instance.get(`${basePath}`);
  }

}

File: back-end/src/main/java/com/ta2khu75/quiz/service/ExamHistoryService.java
package com.ta2khu75.quiz.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.ta2khu75.quiz.entity.request.UserAnswerRequest;
import com.ta2khu75.quiz.entity.response.ExamHistoryResponse;
import com.ta2khu75.quiz.entity.response.PageResponse;
import com.ta2khu75.quiz.entity.response.details.ExamHistoryDetailsResponse;
import com.ta2khu75.quiz.repository.ExamHistoryRepository;

public interface ExamHistoryService {
	ExamHistoryResponse readByExamId(Long id);
	ExamHistoryDetailsResponse scoreByExamId(Long id, Long examId, UserAnswerRequest[] answerUserRequest);
	PageResponse<ExamHistoryResponse> readPage(Pageable pageable);
	ExamHistoryDetailsResponse read(Long id);
}
